Subroutine arguments highlight improvement.

pedro-ricardo opened the pull request:
Stop ending the subroutine `dummy-variable-list` at line break and look for the next `)` to end it.
This fixes issue #70.

It is important to notice that this fix suffers from the same problem as the string markers `'` and `""`.
Now the argument list starts once `(` is typed **in subroutine header** and only ends when the `)` is found.

So, just like the string ones, if you leave the braket open it will keep highlighting. But since Fortran does not allow an open braket on a subroutine I don't believe this is an issue.

### Fix checklist
- [x] Highlight multiple lines in `dummy-variable-list`
- [x] Modify rule to look for `&` character

pedro-ricardo commented:
This is how it looks...
**Before:**
![old](https://user-images.githubusercontent.com/15893711/41512398-21c2f00c-725e-11e8-844a-d4238d78c1c1.png)
**After:**
![new](https://user-images.githubusercontent.com/15893711/41512404-32ec001c-725e-11e8-8fcd-d30098f426ee.png)

pedro-ricardo commented:
Lets make the new release (:+1:)

krvajal commented:
What would happen with invalid code, I think you should try to go to the next line only if the current line ends with `&`

pedro-ricardo commented:
The highlight won't identify that the code is invalid, but linter will.
I agree with you it could be better. Making it look for the `&` was my first try but i could not make it work in all cases. If this `&\n` is typed i can do it, but if typed `&       \n`, it stops working. That's is what i mean by not working in all cases.
If you want this to behave like you said, i'll need more time.

So I vote for leaving this fix out of this new release, since the other ones have direct impact on the extension performance and need to be released as soon as possible.

pedro-ricardo commented:
Latest commit:
* Added back the `\n` as an possible end of the `dummy-variable-list` to avoid the keep highlighting issue
* Added the `line-continuation-operator` to identify `&` and extend `dummy-variable-list`

Now it will only highlight the second line if the first ends with `&` followed by possible blank spaces:
![high](https://user-images.githubusercontent.com/15893711/41947993-1bb0ba84-7991-11e8-8df3-356b8f63a13b.png)
I think that's it

krvajal commented on syntaxes/fortran_free-form.tmLanguage.json:
```diff
@@ -4786,7 +4786,7 @@
 				}
 			},
-			"end": "(\\)|(?=\\n))",
+			"end": "\\)|(?=\\n)",
```
Still not sure if there is a way to check that the line ends with a `&` so the next line is part of the previous

pedro-ricardo replied:
What i've done is simpler, by including the line-continuation-operator in paterns, it prevents the dummy-argument-list to end like that.
In on-line regex websites i found a way to look in the previous line, but it was way more complex than including the & rules we already have in the file.
